src/solvers/CMakeLists.txt

```diff
@@ -65,17 +65,43 @@ list(REMOVE_ITEM sources
 
 add_library(solvers ${sources} ${headers})
 
+include("${CBMC_SOURCE_DIR}/../cmake/DownloadProject.cmake")
+
 if("${sat_impl}" STREQUAL "minisat2")
     message(STATUS "Building solvers with minisat2")
+
+    download_project(PROJ minisat2
+        URL http://ftp.debian.org/debian/pool/main/m/minisat2/minisat2_2.2.1.orig.tar.gz
+        PATCH_COMMAND patch -p1 -i ${CBMC_SOURCE_DIR}/../scripts/minisat-2.2.1-patch
```

Comment:
Is `PATCH_COMMAND` somehow the `CONFIGURE_COMMAND`? Or how does this work?

Reply:
No, the idea is that the project is downloaded and patched at *configure time*, and is then added like a normal subdirectory. This means that there's no need for a dedicated configure command, because the subproject will be configured as part of the superbuild, with all of the same state+flags+environment.

Reply:
Maybe I just don't understand, but my reading of the syntax above is you are calling `download_project` with params `PROJ` as `minisat` `URL` as `http://....` but I can't see a param called `PATCH_COMMAND` in the definition of the `download_project` function (which makes sense - seems a weird thing to do), but how does the `download_project` function know what to do with it. But I know see `PATCH_COMMAND` predates this change so I guess it is being passed along to some later stage.

Reply:
As I understand it, the download_project command intercepts the arguments and passes them directly to a call to `ExternalProject_Add`, overwriting the CONFIGURE_COMMAND and BUILD_COMMAND arguments with empty strings. This is how it knows what to do with the PATCH_COMMAND.